test/endToEnd/workspaceFilteringTest.ts

```diff
@@ -0,0 +1,73 @@
+import {NightwatchBrowser} from 'nightwatch';
+import loginService from './util/loginService';
+import workspaceService, {cleanList} from './util/workspaceService';
+
+export = {
+  afterEach,
+  'Filter workspaces by criteria and alternatives': filterByBoth
+};
+
+function afterEach(browser: NightwatchBrowser) {
+  cleanList(browser);
+}
+
+const getReal = 'GetReal course LU 4, activity 4.4';
+const tervonen =
+  'Antidepressants - single study B/R analysis (Tervonen et al, Stat Med, 2011)';
+const hansen =
+  'Antidepressants - relative effectiveness analysis (Van Valkenhoef et al, J Clin Epi, 2012)';
+
+function filterByBoth(browser: NightwatchBrowser) {
+  loginService.login(browser);
+  workspaceService.cleanList(browser);
+  addWorkspaces(browser);
+
+  filterByCriterion(browser, '2-YEAR SURVIVAL');
+  browser.expect.element('#workspace-0').text.to.equal(getReal);
```

Comment:
should test also check whether there's more workspaces returned? count children?